lib/survey_gizmo/resource.rb

```diff
@@ -44,7 +44,7 @@ def all(conditions = {})
           response = nil
 
           while !response || (all_pages && response['page'] < response['total_pages'])
-            conditions[:page] = response ? response['page'] + 1 : 1
+            conditions[:page] = response ? response['page'] + 1 : 1 unless conditions[:page] && !response
```

Comment:
Looks a bit complicated with the low priority unless mixed with a ternary.
Wouldn't this do the same?:
```ruby
conditions[:page] = response ? response['page'] + 1 : conditions.fetch(:page, 1)
```
It looks cleaner to me